components/raftstore/src/store/fsm/apply.rs

```diff
@@ -2987,8 +2987,16 @@ where
         req: &AdminRequest,
     ) -> Result<(AdminResponse, ApplyResult<EK::Snapshot>)> {
         let is_in_flashback = req.get_cmd_type() == AdminCmdType::PrepareFlashback;
-        // Modify the region meta in memory.
         let mut region = self.region.clone();
+        // First time to enter or leave the flashback state, record the metrics.
+        if region.is_in_flashback != is_in_flashback {
+            if is_in_flashback {
+                PEER_IN_FLASHBACK_STATE.inc();
+            } else {
+                PEER_IN_FLASHBACK_STATE.dec();
+            }
+        }
```

Comment:
What if peer is merged or destroyed? Will they leak the metrics?

Reply:
Once a region is in the flashback state, there is no other raft command that could be proposed and applied, so it's ok to simply inc and dec the counter here.

Reply:
LGTM, though there are some extreme cases that may leak the metrics, eg, a peer is isolated after preparing flashback, and rejoin cluster after it's been removed from raft group via confchange, in this case this peer may be destroyed without applying finish flashback.